unity-renderer/Assets/Rendering/UI/ChargeWheel/UILoadingSpinner.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class UILoadingSpinner : MonoBehaviour
+{
+    Material _material;
+    Image _img;
+
+    public Color color;
+    public float head;
+    public float tail;
+
+    void Start()
+    {
+        GetMaterial();
+    }
+
+    void GetMaterial()
+    {
+        _img = GetComponent<Image>();
+
+        if (_img.maskable)
+        {
+            _material = new Material(_img.material);
+            _material = _img.materialForRendering;
```

Comment:
Why are you doing here the `new Material` assignation? As I can see, the next instruction is already setting the `_material` variable.

```suggestion
            _material = _img.materialForRendering;
```